pkg/network/usm/monitor_test.go

```diff
@@ -872,14 +883,15 @@ func (s *USMHTTP2Suite) TestHTTP2KernelTelemetry() {
 			assert.Eventually(t, func() bool {
 				telemetry, err = usmhttp2.Spec.Instance.(*usmhttp2.Protocol).GetHTTP2KernelTelemetry()
 				require.NoError(t, err)
-
-				return telemetry.Request_seen == tt.expectedTelemetry.Request_seen &&
-					telemetry.Response_seen == tt.expectedTelemetry.Response_seen &&
-					telemetry.Path_exceeds_frame == tt.expectedTelemetry.Path_exceeds_frame &&
-					telemetry.Exceeding_max_interesting_frames == tt.expectedTelemetry.Exceeding_max_interesting_frames &&
-					telemetry.Exceeding_max_frames_to_filter == tt.expectedTelemetry.Exceeding_max_frames_to_filter &&
-					telemetry.End_of_stream+telemetry.End_of_stream_rst >= expectedEOSOrRST &&
-					reflect.DeepEqual(telemetry.Path_size_bucket, tt.expectedTelemetry.Path_size_bucket)
+				require.Equal(t, telemetry.Request_seen, tt.expectedTelemetry.Request_seen)
+				require.Equal(t, telemetry.Response_seen, tt.expectedTelemetry.Response_seen)
+				require.Equal(t, telemetry.Path_exceeds_frame, tt.expectedTelemetry.Path_exceeds_frame)
+				require.Equal(t, telemetry.Exceeding_max_interesting_frames, tt.expectedTelemetry.Exceeding_max_interesting_frames)
+				require.Equal(t, telemetry.Exceeding_max_frames_to_filter, tt.expectedTelemetry.Exceeding_max_frames_to_filter)
+				require.GreaterOrEqual(t, telemetry.End_of_stream+telemetry.End_of_stream_rst, expectedEOSOrRST)
+				require.True(t, reflect.DeepEqual(telemetry.Path_size_bucket, tt.expectedTelemetry.Path_size_bucket))
```

Comment:
this is problematic, as you'll fail the test immediately, instead of using the `Eventually` method

Reply:
I do feel that it is much more convenient in order to understand what is the current issue, maybe I will change the `Eventually` as well?

Reply:
it might be more convenient to read, but it means that you loose the functionality of require.Eventually (meant to deal with races)

you can change to
```
if telemetry.Request_seen != tt.expectedTelemetry.Request_seen {
    return false
}
if telemetry.Response_seen != tt.expectedTelemetry.Response_seen {
    return false
}
...
```

Reply:
I see, changed